### PR TITLE
Feature&Bugs/#5899#6156 #6164/Default display in "set start time" field current time + one hour is not shown. "All day" option doesn't set 00 am to 12 pm. The end time of the event disappears when the userre-select the "All Day" check box 

### PR DESCRIPTION
**Pull request by sobkiv**
Made default display in "set start time" field current time + one hour. By default date is current date
[#5899](https://github.com/ita-social-projects/GreenCity/issues/5899)
[#6156](https://github.com/ita-social-projects/GreenCity/issues/6156)
[#6164](https://www.google.com/url?q=https://github.com/ita-social-projects/GreenCity/issues/6164&sa=D&source=editors&ust=1695907308316125&usg=AOvVaw2LFR_DwyDNy72LHFFtX6SQ)

### CODE REVIEW

File: src/app/main/component/events/components/event-date-time-picker/event-date-time-picker.component.ts [hunk 1]
```diff
@@ -111,6 +115,21 @@ export class EventDateTimePickerComponent implements OnInit, OnChanges {
     }
   }
 
+  private initialStartTime(): any {
+    let initialDate: Date;
+    let initialStartTime = '';
+    if (this.firstFormIsSucceed) {
+      initialDate = new Date();
+      const currentHour = new Date().getHours();
+      if (currentHour + 1 !== 24) {
```
Comment (hnativlyubomyr) [inline src/app/main/component/events/components/event-date-time-picker/event-date-time-picker.component.ts]: use ternary operator here
 initialStartTime = currentHour + 1 !== 24 ? `${currentHour + 1}:00` :  '23:00'

File: src/app/main/component/events/components/event-date-time-picker/event-date-time-picker.component.ts [hunk 2]
```diff
@@ -111,6 +115,21 @@ export class EventDateTimePickerComponent implements OnInit, OnChanges {
     }
   }
 
+  private initialStartTime(): any {
```
Comment (hnativlyubomyr) [inline src/app/main/component/events/components/event-date-time-picker/event-date-time-picker.component.ts]: why any? create interface

File: src/app/main/component/events/components/event-date-time-picker/event-date-time-picker.component.ts [hunk 3]
```diff
@@ -284,6 +310,8 @@ export class EventDateTimePickerComponent implements OnInit, OnChanges {
     const checkTime = time.split(':');
```
Comment (hnativlyubomyr) [inline src/app/main/component/events/components/event-date-time-picker/event-date-time-picker.component.ts]: if (!curTime && checkTime[0] === '00') return
this.timeArrStart = curTime ? [...this.timeArr.slice(curTime + 1, +checkTime[0])] : [...this.timeArr.slice(0, +checkTime[0])]

File: src/app/main/component/events/components/event-date-time-picker/event-date-time-picker.component.ts [hunk 4]
```diff
@@ -302,16 +330,26 @@ export class EventDateTimePickerComponent implements OnInit, OnChanges {
...
+  public canUpdateTimeArrays(): void {
+    this.checkAllDay = false;
+    const startTime = this.dateForm.get('startTime').value;
+    const endTime = this.dateForm.get('endTime').value;
+    this.updateTimeArrays(startTime, endTime);
+  }
+
+  private updateTimeArrays(startTime: string, endTime: string): void {
+    this.fillTimeArray();
+    if (!this.checkAllDay) {
```
Comment (hnativlyubomyr) [inline src/app/main/component/events/components/event-date-time-picker/event-date-time-picker.component.ts]: if (this.checkAllDay) return;

if (this.checkDay()) {
...
} else {
...
}

**Review by KKatiaa — APPROVED** [review]

**Review by hnativlyubomyr — APPROVED** [review]

**Review by Tadei-Mykola — APPROVED** [review]

**Review by NadiiaStoiko — APPROVED** [review]

**Review by OksanaFilipchuk — APPROVED** [review]

**Review by mdashko — APPROVED** [review]

File: src/app/main/component/events/components/event-date-time-picker/event-date-time-picker.component.ts [hunk 5]
```diff
@@ -168,20 +184,24 @@ export class EventDateTimePickerComponent implements OnInit, OnChanges {
 
   public checkIfAllDay(): void {
     this.checkTime = !this.checkTime;
+    this.checkAllDay = true;
```
Comment (sobkiv) [inline src/app/main/component/events/components/event-date-time-picker/event-date-time-picker.component.ts:187]: 6156

File: src/app/main/component/events/components/event-date-time-picker/event-date-time-picker.component.ts [hunk 6]
```diff
@@ -270,29 +290,33 @@ export class EventDateTimePickerComponent implements OnInit, OnChanges {
   }
 
   private fillTimeArray(): void {
```
Comment (sobkiv) [inline src/app/main/component/events/components/event-date-time-picker/event-date-time-picker.component.ts:292]: 6164

**Pull request merged**